src/Lecture2.hs

```diff
@@ -292,4 +404,45 @@ Write a function that takes and expression and performs "Constant
 Folding" optimization on the given expression.
 -}
 constantFolding :: Expr -> Expr
-constantFolding = error "TODO"
```

Comment:
This admittedly went a bit (a lot?) wrong.
I feel I'm not making the most of recursion but my head exploded trying to find a pattern.
Result, still 2 test cases failing, and an awful lot to simplify. :exploding_head: